chia/wallet/dao_wallet/dao_wallet.py

```diff
@@ -0,0 +1,2240 @@
+            cat_wallet_id=uint32(0),
+            dao_cat_wallet_id=uint32(0),
+            proposals_list=[],
+            parent_info=[],
+            current_treasury_coin=None,
+            current_treasury_innerpuz=None,
+            singleton_block_height=uint32(0),
+            filter_below_vote_amount=filter_amount,
+            assets=[],
+            current_height=uint64(0),
+        )
+        info_as_string = json.dumps(self.dao_info.to_json_dict())
+        self.wallet_info = await wallet_state_manager.user_store.create_wallet(
+            name, WalletType.DAO.value, info_as_string
+        )
+        await self.resync_treasury_state()
+        await self.wallet_state_manager.add_new_wallet(self)
+        await self.save_info(self.dao_info)
+        if self.wallet_info is None:
+            raise ValueError("Internal Error")
+        self.wallet_id = self.wallet_info.id
+
+        # Now the dao wallet is created we can create the dao_cat wallet
+        cat_wallet: CATWallet = self.wallet_state_manager.wallets[self.dao_info.cat_wallet_id]
+        cat_tail = cat_wallet.cat_info.limitations_program_hash
+        new_dao_cat_wallet = await DAOCATWallet.get_or_create_wallet_for_cat(
+            self.wallet_state_manager, self.standard_wallet, cat_tail.hex()
+        )
+        dao_cat_wallet_id = new_dao_cat_wallet.wallet_info.id
+        dao_info = dataclasses.replace(self.dao_info, cat_wallet_id=cat_wallet.id(), dao_cat_wallet_id=dao_cat_wallet_id)
+        await self.save_info(dao_info)
+
+        # add treasury id to interested puzzle hashes. This is hinted in funding coins so we can track them
+        await self.wallet_state_manager.add_interested_puzzle_hashes([self.dao_info.treasury_id], [self.id()])
+        return self
+
+    @staticmethod
+    async def create(
+        wallet_state_manager: Any,
+        wallet: Wallet,
+        wallet_info: WalletInfo,
+        name: Optional[str] = None,
+    ) -> DAOWallet:
+        """
+        Create a DID wallet based on the local database
+        :param wallet_state_manager: Wallet state manager
+        :param wallet: Standard wallet
+        :param wallet_info: Serialized WalletInfo
+        :param name: Wallet name
+        :return:
+        """
+        self = DAOWallet()
+        self.log = logging.getLogger(name if name else __name__)
+        self.wallet_state_manager = wallet_state_manager
+        self.wallet_info = wallet_info
+        self.wallet_id = wallet_info.id
+        self.standard_wallet = wallet
+        self.wallet_info = wallet_info
+        self.dao_info = DAOInfo.from_json_dict(json.loads(wallet_info.data))
+        self.dao_rules = get_treasury_rules_from_puzzle(self.dao_info.current_treasury_innerpuz)
+        return self
+
+    @classmethod
+    def type(cls) -> WalletType:
+        return WalletType.DAO
+
+    def id(self) -> uint32:
+        return self.wallet_info.id
+
+    def puzzle_for_pk(self, pubkey: G1Element) -> Program:
```

Comment:
why do we need this method since it's a constant?

Reply:
I thought these were necessary for wallet integration, but seems like they aren't so have removed `puzzle_for_pk`, `puzzle_hash_for_pk`, and `get_new_puzzle`